docs/runner.md

```diff
@@ -0,0 +1,239 @@
+
+1. [Quickstart](#quickstart)
+2. [Key Concepts](#key-concepts)
+    - [Tasks](#tasks)
+    - [Actions](#actions)
+        - [Task](#task)
+        - [Cmd](#cmd)
+    - [Variables](#variables)
+    - [Files](#files)
+    - [Wait](#wait)
+
+## Quickstart
+
+Create a file called `tasks.yaml`
+
+```yaml
+variables:
+  - name: FOO
+    default: foo
+
+tasks:
+  - name: example
+    actions:
+      - task: set-variable
+      - task: echo-variable
+
+  - name: set-variable
+    actions:
+      - cmd: echo "bar"
+        setVariables:
+          - name: FOO
+
+  - name: echo-variable
+    actions:
+      - cmd: echo ${FOO}
+```
+
+From the same directory as the `tasks.yaml`, run the `example` task using:
+
+```
+uds run example
+```
+
+This will run the `example` tasks which in turn runs the `set-variable` and `echo-variable`. In this example, the text "
+bar" should be printed to the screen twice.
+
+Optionally, you can specify the location and name of your `tasks.yaml` using the `--file` or `-f` flag:
+
+```
+uds run example -f tmp/tasks.yaml
+```
+
+## Key Concepts
+
+### Tasks
+
+Tasks are the fundamental building blocks of the UDS runner and they define operations to be performed. The `tasks` key
+at the root of `tasks.yaml` define a list of tasks to be run. This underlying operations performed by a task are defined
+under the `actions` key:
+
+```yaml
+tasks:
+  - name: all-the-tasks
+    actions:
+      - task: make-build-dir
+      - task: install-deps
+```
+
+In this example, the name of the task is "all-the-tasks", and it is composed multiple sub-tasks to run. These sub-tasks
+would also be defined in the list of `tasks`:
+
+```yaml
+tasks:
+  - name: all-the-tasks
+    actions:
+      - task: make-build-dir
+      - task: install-deps
+
+  - name: make-build-dir
+    actions:
+      - cmd: mkdir -p build
+
+  - name: install-deps
+    actions:
+      - cmd: go mod tidy
+```
+
+Using the UDS CLI, these tasks can be run individually:
+
+```
+uds run all-the-tasks   # runs all-the-tasks, which calls make-build-dir and install-deps
+uds run make-build-dir  # only runs make-build-dir
+```
+
+### Actions
+
+Actions are the underlying operations that a task will perform. Each action under the `actions` key has a unique syntax.
+
+#### Task
+
+A task can reference a task, thus making tasks composable.
+
+```yaml
+tasks:
+  - name: foo
+    actions:
+      - task: bar
+  - name: bar
+    actions:
+      - task: baz
+  - name: baz
+    actions:
+      - cmd: "echo task foo is composed of task bar which is composed of task baz!"
+```
+
+In this example, the task `foo` calls a task called `bar` which calls a task `baz` which prints some output to the
+console.
+
+#### Cmd
+
+Actions can run arbitrary bash commands including in-line scripts, and the output of a command can be placed in a
+variable using the `setVariables` key
+
+```yaml
+tasks:
+  - name: foo
+    actions:
+      - cmd: uname -m | sed 's/x86_64/amd64/;s/aarch64/arm64/'
```

Comment:
now that #160 is merged, maybe we should do some different example so ppl aren't confused and try to use that somewhere vs the built-in env variable?

Reply:
Updated with `echo -n 'dHdvIHdlZWtzIG5vIHByb2JsZW0=' | base64 -d`